### PR TITLE
[WFCORE-3825] standardize setting of '-server' option in domain and s…

### PR DESCRIPTION
**Pull request by rsearls**
…tandalone scripts .sh, .bat, .ps1

### CODE REVIEW

**Comment by jmesnil** [comment]
@marekkopecky coud you review please?

File: core-feature-pack/src/main/resources/content/bin/standalone.conf [hunk 1]
```diff
@@ -50,23 +50,24 @@ fi
...
-   JAVA_OPTS="-Xms64m -Xmx512m -XX:MetaspaceSize=96M -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true"
-   JAVA_OPTS="$JAVA_OPTS -Djboss.modules.system.pkgs=$JBOSS_MODULES_SYSTEM_PKGS -Djava.awt.headless=true"
-else
-   echo "JAVA_OPTS already set in environment; overriding default settings with values: $JAVA_OPTS"
-fi
+    JAVA_OPTS="-Xms64m -Xmx512m -XX:MetaspaceSize=96M -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true"
+    JAVA_OPTS="$JAVA_OPTS -Djboss.modules.system.pkgs=$JBOSS_MODULES_SYSTEM_PKGS -Djava.awt.headless=true"
+
+    # Sample JPDA settings for remote socket debugging
+    #JAVA_OPTS="$JAVA_OPTS -agentlib:jdwp=transport=dt_socket,address=8787,server=y,suspend=n"
```
Comment (jmesnil) [inline core-feature-pack/src/main/resources/content/bin/standalone.conf]: that's a change in the script behaviour.
Now the debugging parameters are added only when JAVA_OPTS is not set in the environment.

File: core-feature-pack/src/main/resources/content/bin/domain.conf.ps1 [hunk 2]
```diff
@@ -25,33 +25,34 @@ if (-Not(test-path env:JBOSS_MODULES_SYSTEM_PKGS )) {
   $JBOSS_MODULES_SYSTEM_PKGS="org.jboss.byteman"
 }
 
+# Set default values if none have been set by the user
```
Comment (jmesnil) [inline core-feature-pack/src/main/resources/content/bin/domain.conf.ps1]: How is that change related to the `-server` unification?

Comment (rsearls) [reply core-feature-pack/src/main/resources/content/bin/domain.conf.ps1]: The change to standalone.conf.ps1 does not change the handling of JAVA_OPTS
by standalone.ps1.  Function Get-Java-Arguments in common.ps1 calls function
Get-Java-Opts.  This function checks the setting of PRESERVE_JAVA_OPTS and
JAVA_OPTS.  It chooses the user's JAVA_OPTS if set or the values set in
standalone.conf.ps1.

**Comment by rsearls** [comment]
@jmesmil  Why is this PR being held up?   I believe I have sufficiently answered your question.

File: core-feature-pack/src/main/resources/content/bin/standalone.conf (hunk 1 above)
Comment (jmesnil) [reply core-feature-pack/src/main/resources/content/bin/standalone.conf]: @rsearls there is still this part that needs to be updates, the commented settings for debugging should be after the if statement, not inside it.

Comment (marekkopecky) [reply core-feature-pack/src/main/resources/content/bin/standalone.conf]: @jmesnil +1, this is regression of behaviour, I recommend to remove all changes of standalone.conf file.

File: core-feature-pack/src/main/resources/content/bin/domain.conf [hunk 3]
```diff
@@ -48,13 +48,14 @@ fi
 if [ "x$JAVA_OPTS" = "x" ]; then
    JAVA_OPTS="-Xms64m -Xmx512m -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true"
    JAVA_OPTS="$JAVA_OPTS -Djboss.modules.system.pkgs=$JBOSS_MODULES_SYSTEM_PKGS -Djava.awt.headless=true"
+
+   # Use JBoss Modules lockless mode
```
Comment (marekkopecky) [inline core-feature-pack/src/main/resources/content/bin/domain.conf]: Moving of these commented JAVA_OPTS settings is not described in [jira](https://issues.jboss.org/browse/WFCORE-3825) or in description of this PR. These changes could cause the regression of behaviour. I recommend to remove all changes of domain.conf file.

File: core-feature-pack/src/main/resources/content/bin/domain.sh [hunk 4]
```diff
@@ -131,8 +135,6 @@ if [ "x$SERVER_SET" = "x" ]; then
             JVM_OPTVERSION="-server $JVM_OPTVERSION"
         fi
     fi
-else
```
Comment (marekkopecky) [inline core-feature-pack/src/main/resources/content/bin/domain.sh]: Why this else block needs to be removed?

Anyway, [original jira](https://issues.jboss.org/browse/WFCORE-3825) is just about windows scripts, if you want to change domain.sh script, please describe it in [original jira](https://issues.jboss.org/browse/WFCORE-3825) description and PR description or create new jira and PR for that.

File: core-feature-pack/src/main/resources/content/bin/domain.bat [hunk 5]
```diff
@@ -78,6 +78,43 @@ if "%OS%" == "Windows_NT" (
...
+         if errorlevel == 0 (
+                set JVM_OPTVERSION="-version"
+          )
+    )
+)
+
+echo "%JAVA_OPTS%" | findstr /I \-server > nul
+if errorlevel == 1 (
+
+  echo "%JAVA%" "%JVM_OPTVERSION%" 2>&1 | findstr /I /C:"HotSpot" > nul
```
Comment (marekkopecky) [inline core-feature-pack/src/main/resources/content/bin/domain.bat]: HotSpot detection is currently on line 103 (https://github.com/wildfly/wildfly-core/blob/master/core-feature-pack/src/main/resources/content/bin/domain.bat#L103), this `-server` settings is used on HC, PC and servers. Is this code duplication?

File: core-feature-pack/src/main/resources/content/bin/standalone.bat [hunk 6]
```diff
@@ -42,6 +42,7 @@ if not errorlevel == 1 (
 )
 setlocal DisableDelayedExpansion
 
+set outParam=
```
Comment (marekkopecky) [inline core-feature-pack/src/main/resources/content/bin/standalone.bat]: [Original jira](https://issues.jboss.org/browse/WFCORE-3825) is just about domain scripts. Please update description of original jira, or move this fixes to different PR and different jira.

File: core-feature-pack/src/main/resources/content/bin/standalone.conf (hunk 1 above)
Comment (rsearls) [reply core-feature-pack/src/main/resources/content/bin/standalone.conf]: All standalone.* file changes withdrawn.   As marekkopecky recommends create an new jira for these files.

File: core-feature-pack/src/main/resources/content/bin/domain.bat (hunk 5 above)
Comment (rsearls) [reply core-feature-pack/src/main/resources/content/bin/domain.bat]: Lines 84-100 are checking for text "-d64" and "-d32".
  Lines 103-114 are checking for "HotSpot" or "OpenJDK"
  There seems to be some combination of -d64 HotSpot, -d32 HotSpot,
  -d64 OpenJDK, -d32 OpenJdk that must be identified on all platforms.

Comment (marekkopecky) [reply core-feature-pack/src/main/resources/content/bin/domain.bat]: > Lines 103-114 are checking for "HotSpot" or "OpenJDK"

Ok ... so are you agree with me, that this hotspot detection is a code duplication? 

See [original line 103](https://github.com/wildfly/wildfly-core/blob/master/core-feature-pack/src/main/resources/content/bin/domain.bat#L103):

```
 "%JAVA%" -server -version 2>&1 | findstr /I hotspot > nul
if not errorlevel == 1 (
...
```

and compare it with your new code:

```
 echo "%JAVA%" "%JVM_OPTVERSION%" 2>&1 | findstr /I /C:"HotSpot" > nul
  if errorlevel == 0 (
```

Comment (marekkopecky) [reply core-feature-pack/src/main/resources/content/bin/domain.bat]: Btw, is the "echo" command really needed in your code?

Comment (marekkopecky) [reply core-feature-pack/src/main/resources/content/bin/domain.bat]: @jamezp Can you please review this PR as well?

Comment (jamezp) [reply core-feature-pack/src/main/resources/content/bin/domain.bat]: @marekkopecky I'll see if I can find the time. I'm stretched pretty thin ATM :)

Comment (rsearls) [reply core-feature-pack/src/main/resources/content/bin/domain.bat]: Yes it is.   This how var contents is pass to the function findstr.

Comment (rsearls) [reply core-feature-pack/src/main/resources/content/bin/domain.bat]: The task (lines 140-144) is performing is NOT a duplicate or (lines 102-114)
At line 102 if the user does not have "-server" set in JAVA_OPTS then 
that if block is executed.   When "-server" is already set in JAVA_OPTS
the "if" block (lines 103-114) is skipped.  At (lines 140-144) we must 
determine if "hotspot" is set and set the 2 variables accordingly.  This 
block of code is not a duplicate needs to remain as is.

**Comment by jamezp** [comment]
retest this please

**Comment by marekkopecky** [comment]
@jamezp Original jira has no assignee now. Can you close this PR?

**Comment by jamezp** [comment]
Good call @marekkopecky
